libraries/AP_HAL/examples/RCInput/RCInput.cpp

```diff
@@ -34,7 +34,7 @@ void loop(void)
         }
     }
     if (changed) {
-        for (uint8_t i=0; i<max_channels; i++) {
+        for (uint8_t i=0; i<=max_channels; i++) {
```

Comment:
It doesn't seem right. See that in the printf() below we do `i + 1` to be 1-based instead of zero-based. With this change we are possibly accessing last_value[MAX_CHANNELS], which is off by one in the array size.